clusterloader/framework/util.go

```diff
@@ -21,19 +21,55 @@ import (
 	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
 const maxRetries = 5
 
+// CreateNS creates a namespace if it is new, otherwise it will return the existing namespace pointer
+func CreateNS(f *framework.Framework, namespaceName string) (*v1.Namespace, error) {
+	var ns *v1.Namespace
+	var err error
+	fullNamespace := getNamespace(f, namespaceName)
+
+	if fullNamespace == "" {
+		ns, err = f.CreateNamespace(namespaceName, nil)
+		if err != nil {
+			return nil, err
+		}
+		framework.Logf("Created new namespace: %s", namespaceName)
+	} else {
+		ns, err = f.ClientSet.CoreV1().Namespaces().Get(fullNamespace, metav1.GetOptions{})
```

Comment:
Since it is getting namespace in case it already exists, CreateNS is not the best name for this function, I guess.

BTW - what's the usecase for it? When we already have a namespace?